### PR TITLE
Add an ARCHITECTURE.md.

### PR DESCRIPTION
**Pull request by DanAlbert**

### CODE REVIEW

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+just images for the READMEs and this doc.
+
+### gradle/libs.versions.toml
+
+This is a Gradle [version catalog]. It's the central location that defines the
+library and plugin dependencies for each sample.
+
+The Android Gradle Plugin does not support evaluating version catalog fields in
+its DSL, so versions for SDK and NDK tools (`ndkVersion`, `compileSdkVersion`,
+`targetSdkVersion`, etc) are all defined by the convention plugins in
```
Comment (enh-google) [inline ARCHITECTURE.md:57]: (ah, yeah, maybe this is a thing then... :-) )

File: ARCHITECTURE.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+[build-logic](#build-logic).
+
+[version catalog]: https://docs.gradle.org/current/userguide/platforms.html
+
+### Gradle wrapper
+
+The `gradlew` and `gradlew.bat` scripts are the Gradle wrappers, bash- and
+Windows batch-compatible respectively. The will manage the Gradle installation
```
Comment (enh-google) [inline ARCHITECTURE.md:65]: s/The/They/

File: ARCHITECTURE.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+The Android Gradle Plugin does not support evaluating version catalog fields in
+its DSL, so versions for SDK and NDK tools (`ndkVersion`, `compileSdkVersion`,
+`targetSdkVersion`, etc) are all defined by the convention plugins in
+[build-logic](#build-logic).
+
+[version catalog]: https://docs.gradle.org/current/userguide/platforms.html
+
+### Gradle wrapper
+
+The `gradlew` and `gradlew.bat` scripts are the Gradle wrappers, bash- and
```
Comment (enh-google) [inline ARCHITECTURE.md:64]: maybe "macOS/Linux- and Windows-compatible" is more to the point/intelligible. (i suspect anyone who knows bash from a batch file already knows what they're looking at here!)

File: ARCHITECTURE.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+`gradle/wrapper/gradle-wrapper.properties` defines which version of Gradle will
+be used. This should rarely be modified manually. Android Studio's Upgrade
+Assistant will manage that file and upgrade to whatever version of Gradle it
+prefers for that version of Android Studio and Android Gradle.
+
+### build.gradle and settings.gradle
+
+The build.gradle (or build.gradle.kts) file is the top level build file. It
+doesn't do anything interesting but declare which plugins will be used by the
```
Comment (enh-google) [inline ARCHITECTURE.md:77]: "anything but declare" or "anything interesting but declares"?

File: ARCHITECTURE.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+child modules. Each sample module has its own build.gradle file that defines
```
Comment (enh-google) [inline ARCHITECTURE.md:78]: "sample's module" (since these aren't samples of modules, these are samples that happen to require a module each?)

**Review by enh-google — APPROVED** [review]

File: ARCHITECTURE.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+Android Gradle modules have multiple "build types", sometimes called "variants".
+For most of the samples, there are only two build types: debug and release. A
+few samples, notably the sanitizers sample, have more. To view the build types
+for the modules in this repository, in the Android Studio application menu,
+select View -> Tool Windows -> Build Variants. A window will open that lets you
+select the active variant for each sample.
+
+### build-logic
+
+The `build-logic` directory contains Gradle convention plugins used by this
```
[the remark this answers is not in the document]
Comment (DanAlbert) [reply ARCHITECTURE.md:39]: Yeah: https://docs.gradle.org/current/samples/sample_convention_plugins.html

**Pull request merged**